### PR TITLE
Handle uncompressed layers nicely

### PR DESCRIPTION
**Pull request by kzys**
For uncompressed layers, images.DiffCompression() returns "unknown" but buildSociLayer() was only telling it wasn't gzip.

Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>

*Issue #, if available:*

*Description of changes:*

*Testing performed:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rdpsin — DISMISSED** [review]

File: soci/soci_index_test.go [hunk 1]
```diff
@@ -352,3 +353,15 @@ func TestNewIndexFromReader(t *testing.T) {
...
 }
+
+func TestBuildSociLayer(t *testing.T) {
+	ctx := context.Background()
+	desc := ocispec.Descriptor{
+		MediaType: images.MediaTypeDockerSchema2Layer,
+		Digest:    "layerdigest",
+	}
+	_, err := buildSociLayer(ctx, nil, desc, 0, nil, nil)
+	if err.Error() != `layer layerdigest (application/vnd.docker.image.rootfs.diff.tar) must be compressed by gzip, but got "unknown"` {
```
Comment (wmesard) [inline soci/soci_index_test.go:364]: This is unnecessarily brittle. Tests should protect functionality, not enshrine current behavior. When we add support for Zstandard layers, this error message will change, and the test--as currently written--will break. (I'll admit that my real complaint here is philosophical, not technical. If developers get in the habit of tweaking unit tests, they learn to ignore test failures even when a failure is trying to tell them about a real problem.)

**TL;DR:** Suggest testing only for the behavior that you want to protect. Something like:
```
        if !strings.Contains(err.Error(), "layerdigest") || !strings.Contains(err.Error(), "application/vnd.docker.image.rootfs.diff.tar") ||
                !strings.Contains(err.Error(), "compress") {
                t.Fatalf("unexpected error: %v", err)

```

File: soci/soci_index_test.go [hunk 2]
```diff
@@ -352,3 +353,15 @@ func TestNewIndexFromReader(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildSociLayer(t *testing.T) {
```
Comment (vkuzniet) [inline soci/soci_index_test.go:357]: Since this test tests for one of the error modes in `buildSociLayer`, maybe it may make sense to name it respectively, e.g. `TestBuildSociLayerErrorModes`, or something like that?

File: soci/soci_index_test.go (hunk 1 above)
Comment (kzys) [reply soci/soci_index_test.go:364]: Sounds good. How about returning ErrFailedPrecondition from https://pkg.go.dev/github.com/containerd/containerd/errdefs in addition to?

Comment (kzys) [reply soci/soci_index_test.go:364]: Or define something like errUnsupportedCompression

Comment (rdpsin) [reply soci/soci_index_test.go:364]: Would prefer the latter.

Comment (rdpsin) [reply soci/soci_index_test.go:364]: Something like `errUnsupportedLayerFormat`.

**Review by rdpsin — APPROVED** [review]

**Review by vkuzniet — APPROVED** [review]

**Review by wmesard — APPROVED** [review]
